.github/workflows/build-and-test.yml

```diff
@@ -130,6 +130,22 @@ jobs:
             echo "One or more matrix jobs failed."
             false
           fi
+  govulncheck:
+    runs-on: ubuntu-latest
+    timeout-minutes: 5
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Install Go
+        uses: actions/setup-go@v3
+        with:
+          cache: true
+          check-latest: true
+          go-version: ~1.19.7
+      - name: Install `govulncheck`
+        run: go install golang.org/x/vuln/cmd/govulncheck@latest
+      - name: Run `govulncheck`
+        run: govulncheck ./...
```

Comment:
I like this approach more than others.

Q: Is this applied to all the modules? If so, we need to remove or update the make target

Reply:
I'll double check but this should descend recursively to all packages via the `./...`

Yes, the makefile stuff is to be removed if we choose this approach.

Reply:
> I like this approach more than others.

So you're voting for option 2 (I prefer 1 for a more consistent approach to the rest of the checks there).
Do you want to go with option 2 per your choice or do we want to wait for some additional feedback as well?

Reply:
Isn't this 1 (as a job in build-and-test workflow)?

Reply:
omg, it is, sorry :D I did not sleep well this night ;) ok, so I'll clean this up and prepare for merging, and again, sorry for the confusion.

Reply:
ok, cleanup done.

And I rewrote this a bit so that it actually uses the `make govulncheck` target right now. I think this is better this way because:
* we have a clean versioning approach via `tools` module
* it's consistent with the rest of scans around